cli/commands/catalog/module/repo.go

```diff
@@ -137,36 +136,49 @@ func (repo *Repo) moduleURL(moduleDir string) (string, error) {
 
 // clone clones the repository to a temporary directory if the repoPath is URL
 func (repo *Repo) clone(ctx context.Context) error {
-	if repo.path == "" {
+	if repo.cloneURL == "" {
 		currentDir, err := os.Getwd()
 		if err != nil {
 			return errors.WithStackTrace(err)
 		}
 
-		repo.path = currentDir
+		repo.cloneURL = currentDir
 	}
 
-	if files.IsDir(repo.path) {
-		if !filepath.IsAbs(repo.path) {
-			absRepoPath, err := filepath.Abs(repo.path)
+	if repoPath := repo.cloneURL; files.IsDir(repoPath) {
+		if !filepath.IsAbs(repoPath) {
+			absRepoPath, err := filepath.Abs(repoPath)
 			if err != nil {
 				return errors.WithStackTrace(err)
 			}
 
-			log.Debugf("Converting relative path %q to absolute %q", repo.path, absRepoPath)
-
-			repo.path = absRepoPath
+			log.Debugf("Converting relative path %q to absolute %q", repoPath, absRepoPath)
 		}
+		repo.path = repoPath
 
 		return nil
 	}
 
-	if err := os.MkdirAll(repo.tempDir, os.ModePerm); err != nil {
+	if err := os.MkdirAll(repo.path, os.ModePerm); err != nil {
 		return errors.WithStackTrace(err)
 	}
-	repo.tempDir = filepath.Join(repo.tempDir, "temp")
 
-	sourceUrl, err := terraform.ToSourceUrl(repo.cloneUrl, "")
+	repoName := "temp"
+	if match := repoNameFromCloneURLReg.FindStringSubmatch(repo.cloneURL); len(match) > 0 && match[1] != "" {
+		repoName = match[1]
+	}
+
+	repo.path = filepath.Join(repo.path, repoName)
+
+	if files.FileExists(repo.path) && !files.FileExists(repo.gitHeadfile()) {
+		log.Debugf("The repo dir exists but git file %q does not. Removing the repo dir for cloning from the remote source.", repo.gitHeadfile())
+
+		if err := os.RemoveAll(repo.path); err != nil {
+			return errors.WithStackTrace(err)
+		}
+	}
```

Comment:
NIT: please add a comment explaining what you're checking for here and why it's necessary (i.e., tmp folders being deleted by macOs). You could link back to this PR too for more context.

Reply:
Fair point. Added the comment.